src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -734,130 +740,714 @@ public object InvokeFunction(string methodName, Query query, string documentId,
             }
         }
 
-        private BlittableJsonReaderObject InvokeTimeSeriesFunction(TimeSeriesFunction func, string documentId, object[] args)
+        private BlittableJsonReaderObject InvokeTimeSeriesFunction(DeclaredFunction declaredFunction, string documentId, object[] args)
```

Comment:
can we make this class partial and move all TS stuff to `QueryResultRetrieverBase..TimeSeries.cs` ? This becomes monolith.

Reply:
Or maybe we can extract that to separate class and introduce field e.g. `TimeSeries` in this retriever?

Reply:
I like this idea, yes

Reply:
done